### ISSUE TITLE
l2_reg_loss_val is going to NAN

### ISSUE THREAD

**Issue by malfonsoNeoris** [alexander-pv/maskrcnn_tf2#2]
hi, its mi again! LoL
im having some problems ( i dunno if this is expected or not) and the training sessions is failing after some epochs

for giving you more details.. i have a small dataset of 500 license plates images ( with some padding) from which around 400 are for training.

im doing some training with different backbones, and img size.. to reduce memory footprint and increase FPS.

im getting this error after some epochs
its seems that the l2_reg_loss_val is going to NAN.. have you experienced this ?
When happens.. depends on the image size.
Example keeping mobilenet default and changing the size
512 epoch 10
256 epoch 2

for example for a resnet18 256x256 epoc 6

-------------------HERE THE ERROR--------------
  File "src/plates_train.py", line 68, in <module>
    train_model(model,
  File "/src/src/training.py", line 122, in train_model
    model.fit(train_datagen.repeat(),
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/training.py", line 1100, in fit
    tmp_logs = self.train_function(iterator)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/training.py", line 805, in train_function
    return step_function(self, iterator)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/training.py", line 795, in step_function
    outputs = model.distribute_strategy.run(run_step, args=(data,))
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/distribute/distribute_lib.py", line 1259, in run
    return self._extended.call_for_each_replica(fn, args=args, kwargs=kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/distribute/distribute_lib.py", line 2730, in call_for_each_replica
    return self._call_for_each_replica(fn, args, kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/distribute/distribute_lib.py", line 3417, in _call_for_each_replica
    return fn(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/autograph/impl/api.py", line 572, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/keras/engine/training.py", line 788, in run_step
    outputs = model.train_step(data)
  File "/src/src/model.py", line 91, in train_step
    assert not np.any(np.isnan(l2_reg_loss_val))
AssertionError

**Comment by alexander-pv** [comment]
Hi, @malfonsoNeoris,

I had such a problem earlier, and its cause was an excess of augmentation.
Probably, a softer augmentation is needed for your dataset. The basic augmentation is stored in `src/preprocess/augmentation.py`, `get_training_augmentation` function.

You may save and view the augmented images to see what can be wrong. You can also try to comment out the entire augmentation in `get_training_augmentation` except flipping and rotating operations and then recursively add others image transform operations back to detect which operation produces nans for your dataset.

**Comment by malfonsoNeoris** [comment]
hi alexander.
so i did a soft augmenter, just comment some of the lines of augmentations and the trainning completed succesfully.
( now i have another problem.. in new issue)

**Comment by alexander-pv** [comment]
Ok. Closing the issue.
